pkg/controller/host/validation.go

```diff
@@ -133,20 +136,32 @@ func (r *Reconciler) validateID(host *api.Host) error {
 	default:
 		return nil
 	}
-	status, err := inventory.Get(resource, host.Spec.ID)
+	err = inventory.Find(resource, ref)
 	if err != nil {
+		if errors.Is(err, web.NotFoundErr) {
+			host.Status.SetCondition(
+				libcnd.Condition{
+					Type:     HostNotValid,
+					Status:   True,
+					Reason:   NotFound,
+					Category: Critical,
+					Message:  "Referenced host not found.",
+				})
+			return nil
+		}
+		if errors.Is(err, web.RefNotUniqueErr) {
+			host.Status.SetCondition(
+				libcnd.Condition{
+					Type:     HostNotValid,
+					Status:   True,
+					Reason:   Ambiguous,
+					Category: Critical,
+					Message:  "Host reference is ambiguous.",
+				})
+			return nil
+		}
 		return liberr.Wrap(err)
 	}
-	if status != http.StatusOK {
```

Comment:
The web.Client methods updated to return errors for not-found and ambiguous find result.